packages/web-app-files/src/components/FilesList/ResourceTiles.vue

```diff
@@ -158,44 +151,52 @@ export default defineComponent({
       }
     }
 
-    const contextMenuLabel = computed(() => {
-      return $gettext('Show context menu')
-    })
-
-    const resetDropPosition = (event, index, reference) => {
-      const drop = tileRefs.value.dropEls[index].tippy
-
-      if (drop === undefined) {
+    const showContextMenuOnBtnClick = (data, item, index) => {
+      const { dropdown, event } = data
+      if (dropdown?.tippy === undefined) {
         return
       }
-      displayPositionedDropdown(drop, event, reference)
+      displayPositionedDropdown(dropdown.tippy, event, unref(tileRefs).dropBtns[index])
     }
 
     const showContextMenu = (event, index, reference) => {
       event.preventDefault()
-      const drop = tileRefs.value.dropEls[index]
+      const drop = unref(tileRefs).tiles[index]?.$el.getElementsByClassName(
+        'resource-tiles-btn-action-dropdown'
+      )[0]
 
       if (drop === undefined) {
         return
       }
-      displayPositionedDropdown(drop.tippy, event, reference)
+      displayPositionedDropdown(drop._tippy, event, reference)
+    }
+
+    const isResourceSelected = (resource) => {
+      return props.selectedIds.includes(resource.id)
+    }
+
+    const setSelection = (resource) => {
+      const selectedIds = !isResourceSelected(resource)
+        ? [...props.selectedIds, resource.id]
+        : props.selectedIds.filter((id) => id !== resource.id)
+      context.emit('update:selectedIds', selectedIds)
     }
```

Comment:
I think `setSelection` as a name is a bit awkward, if there's no value parameter. 
maybe rather call it `updateSelection` or pass `isResourceSelected(resource)` to the function as value argument.

Reply:
Good point. Apparently we use `toggleSelect...` quite a bit in the project, I guess that's also appropriate here.

Reply:
Yeah, if it's always a toggle - then that's certainly best. I didn't look closely and thought it might be required to be able to call it without changing the state